### ISSUE TITLE
Unable to use Docker on Xubuntu 16.10 if kernel is `4.8.0-34`.

### ISSUE THREAD

**Issue by hickscorp** [docker/docker#30709]
**Description**

On a fresh install of Xubuntu 16.10, and after installing docker.io from vanilla repositories doesn't allow to use docker correctly.

**Steps to reproduce the issue:**
1. Install Xubuntu 16.10 (Most likelly for any variant of Ubuntu 16.10, not tested tho), **with kernel `4.0.8-34`**.
2. `apt-get install docker.io`, add your user to the `docker` group, etc.
3. Use docker / docker-compose.

**Describe the results you received:**

When running `docker-compose up`: `ERROR: containerd: container not started`.
**Please note that the bug disapears on `4.8.0-37`.** See outputs below.

## Infos on `4.8.0-34`

**Output of `docker version`:**

```
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   78d1802
 Built:        Tue Jan 31 23:47:34 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Mon, 19 Dec 2016 09:20:48 +1300
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 1
Server Version: 1.12.3
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 9
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.8.0-34-generic
Operating System: Ubuntu 16.10
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.59 GiB
Name: AirforceOne
ID: CXYO:3JLG:ODX2:TCXW:NHX7:EHKH:QIYD:ODJO:AB62:LJ4U:2WJU:N7Y3
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

## Infos on `4.8.0-37`

**Output of `docker version`:**

```
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   78d1802
 Built:        Tue Jan 31 23:47:34 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   78d1802
 Built:        Tue Jan 31 23:47:34 2017
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 4
 Running: 4
 Paused: 0
 Stopped: 0
Images: 2
Server Version: 1.12.6
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 26
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.8.0-37-generic
Operating System: Ubuntu 16.10
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.59 GiB
Name: AirforceOne
ID: CXYO:3JLG:ODX2:TCXW:NHX7:EHKH:QIYD:ODJO:AB62:LJ4U:2WJU:N7Y3
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

**Comment by hickscorp** [comment]
@GordonTheTurtle I'm sorry, I didn't realize that the bug disapeared on `4.8.0-37`. I have updated the information, please tell me if anything else is needed (I have both kernels so if you need diffing between stuff, tell me).

**Comment by cpuguy83** [comment]
The `docker.io` package is not an official package but rather is provided by Canonical.
Please try with the official packages. `apt-get remove docker.io && curl -SLf https://get.docker.com | sh`

**Comment by cpuguy83** [comment]
Also if you can provide any logs from the daemon it would be extremely helpful.

**Comment by hickscorp** [comment]
@cpuguy83 Thanks for the answer. I'll install the official version and will run tests with the kernel that was showing the issue. Please don't close this issue, it will take me up to 2 days to get access to that computer.

**Comment by hickscorp** [comment]
Sorry for the delay, I just took the oportunity of having some time to do more tests.

@cpuguy83's suggestion indeed works. I just tested again using the `.34`, `.37` and `.38` versions of the kernel, and there was no problem at all.

When I use docker in production, I either let the installation to the host, or have an ansible script to install everything via wget / docker.io. I never actually realized that the ubuntu vanilla packages were any different.

Sorry for the noise!
